plugins/kubernetes/app/models/kubernetes/resource_status.rb

```diff
@@ -40,6 +40,17 @@ def check
         else
           @details = "Waiting (#{@pod.phase}, #{@pod.reason})"
         end
+      elsif kind == "HorizontalPodAutoscaler"
+        hpa = Kubernetes::Api::Hpa.new if @resource
+        failures = hpa.events_indicating_failure(events(type: "Warning"))
```

Comment:
this line will fail if there's no @resource because hpa will be undefined. so combine the @resource if check to the elsif on top

Reply:
could resource be invalid ?anyways that check is not needed here.